feature/floor-map/src/main/java/io/github/droidkaigi/confsched2023/floormap/FloorMapScreen.kt

```diff
@@ -98,53 +106,76 @@ private fun FloorMapScreen(
+                        Text(
+                            text = FloorMapStrings.Title.asString(),
+                            style = MaterialTheme.typography.headlineLarge,
+                            fontWeight = FontWeight.Medium,
+                        )
+                    } else {
+                        Text(
+                            text = FloorMapStrings.Title.asString(),
+                            style = MaterialTheme.typography.titleLarge,
+                            modifier = Modifier.alpha(scrollBehavior.state.overlappedFraction),
+                        )
+                    }
                 },
+                scrollBehavior = scrollBehavior,
             )
         },
         content = { innerPadding ->
+            val density = LocalDensity.current
+            var boxHeight by remember { mutableIntStateOf(0) }
+            var switcherPosition by remember { mutableFloatStateOf(0f) }
+            val gradientStartRatio by remember {
+                derivedStateOf {
+                    if (boxHeight == 0 || switcherPosition == 0f) {
+                        1f
+                    } else {
+                        switcherPosition / (boxHeight - with(density) { 24.dp.toPx() })
+                    }
+                }
+            }
             Box(
                 Modifier
                     .padding(innerPadding)
-                    .padding(horizontal = 16.dp),
+                    .padding(horizontal = 16.dp)
+                    .onGloballyPositioned {
+                        boxHeight = it.size.height
+                    },
             ) {
-                Column(
-                    Modifier
-                        .fillMaxSize(),
-                ) {
-                    FloorMap(uiState = uiState.floorMapUiState)
-                    FloorMapSideEventList(
-                        uiState = uiState.floorMapSideEventListUiState,
-                        onSideEventClick = onSideEventClick,
-                        modifier = Modifier
-                            .weight(1f)
-                            .padding(bottom = 56.dp)
-                            .fadingEdge(
-                                Brush.verticalGradient(
-                                    0.85f to Color.Black,
-                                    1f to Color.Transparent,
-                                ),
+                FloorMapAndSideEventList(
+                    floorMapUiState = uiState.floorMapUiState,
+                    sideEventListUiState = uiState.floorMapSideEventListUiState,
+                    onSideEventClick = onSideEventClick,
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(bottom = 56.dp)
+                        .nestedScroll(scrollBehavior.nestedScrollConnection)
+                        .fadingEdge(
+                            Brush.verticalGradient(
+                                gradientStartRatio to Color.Black,
```

Comment:
To set the edge gradient appropriately regardless of screen size or orientation, the gradient start position is calculated and set based on the floor changeover switch and screen size.